File: frontend/src/types/keci/popup.ts
export type Popup = {
    id: number
    title: string
    imageUrl: string
    repeatable: boolean
    isActive: boolean
    createdAt: string
}

export type CreatePopupRequest = {
    title: string
    repeatable: boolean
    image: File
}

export type UpdatePopupRequest = {
    title: string
    repeatable: boolean
    image?: File
}

export type PopupResponse = {
    id: number
    title: string
    imageUrl: string
    repeatable: boolean
}

File: KeciApp.API/Models/Popup.cs
using System.ComponentModel.DataAnnotations;

namespace KeciApp.API.Models;

public class Popup
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(100)]
    public string Title { get; set; }

    [Required]
    public string ImageUrl { get; set; }

    public bool IsActive { get; set; }

    public bool Repeatable { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
